app/models/embedded_ansible_worker/object_management.rb

```diff
@@ -53,4 +54,74 @@ def ensure_host(provider, connection)
       :variables => {'ansible_connection' => "local"}.to_yaml
     ).id
   end
+
+  def ensure_plugin_playbooks_project_seeded(connection)
+    clean_consolidated_plugin_directory
+    copy_plugin_ansible_content
+
+    commit_git_plugin_content
+
+    if project = existing_plugin_playbook_project(connection)
+      update_playbook_project(project)
+    else
+      create_playbook_project(connection)
+    end
+  ensure
+    # we already have 2 copies: one in the gem and one imported into ansible in the project, delete the temporary one
+    clean_consolidated_plugin_directory
+  end
+
+  private
+
+  CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR = Pathname.new("/var/lib/awx_consolidated_source/projects/gem_ansible_content").freeze
+  def clean_consolidated_plugin_directory
+    FileUtils.rm_rf(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+  end
+
+  def copy_plugin_ansible_content
+    FileUtils.mkdir_p(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+
+    #TODO: make this a public api via an attr_reader
+    Vmdb::Plugins.instance.instance_variable_get(:@registered_ansible_content).each do |content|
+      FileUtils.cp_r(Dir.glob("#{content.path}/*"), CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+    end
+  end
+
+  def commit_git_plugin_content
+    Dir.chdir(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR) do
+      require 'rugged'
+      repo = Rugged::Repository.init_at(".")
+      index = repo.index
+      index.add_all("*")
+      index.write
+
+      options              = {}
+      options[:tree]       = index.write_tree(repo)
+      options[:author]     = options[:committer] = { :email => "user@example.com", :name => 'Author', :time => Time.now }
+      options[:message]    = "YOLO Initial Commit"
+      options[:parents]    = []
+      options[:update_ref] = 'HEAD'
+      Rugged::Commit.create(repo, options)
+    end
+  end
+
+  PLUGIN_PLAYBOOK_PROJECT_NAME = "Default ManageIQ Playbook Project".freeze
+  PLAYBOOK_PROJECT_ATTRIBUTES = {
+      :name                 => PLUGIN_PLAYBOOK_PROJECT_NAME,
+      :scm_type             => "git",
+      :scm_url              => "file:://#{CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR}",
+      :scm_update_on_launch => false
+  }.freeze
+
+  def existing_plugin_playbook_project(connection)
+    connection.api.projects.all(:name => PLUGIN_PLAYBOOK_PROJECT_NAME).first
+  end
+
+  def update_playbook_project(project)
+    project.update_attributes!(PLAYBOOK_PROJECT_ATTRIBUTES)
+  end
+
+  def create_playbook_project(connection)
+    connection.api.projects.create!(PLAYBOOK_PROJECT_ATTRIBUTES.to_json)
```

Comment:
Does anyone else need this id? Should we save it against the provider like we do for the other initial objects?

Reply:
@syncrou and I discussed this briefly and it didn't seem that this project would ever get accessed in manageiq after the fact, which is why we're searching by name for the "update" part.  If we would need schema changes, we can't really backport that though.

Reply:
We wouldn't need schema changes. These are built using custom attributes https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/embedded_ansible/provider/default_ansible_objects.rb

Reply:
ok, @gmcculloug should I capture the project id for this?

Reply:
@syncrou what do you think of this?  I can store it if needed.

Reply:
@jrafanie - I can't think of a reason we need it, if we can rebuild the entire system from the db when enabling the role on another machine (or upgrading ), then it seems to me that id is not needed at all.